File: MODULE 2.0/src/ss00_case_study/furama_resort/models/Customer.java
package ss00_case_study.furama_resort.models;

public class Customer extends Person{
    private String typeCustomer;

    public Customer() {
    }

    public Customer(String typeCustomer) {
        this.typeCustomer = typeCustomer;
    }

    public Customer(int id, String name, String dateOfBirth, String gender, String idCard, String phoneNumber, String email, String address, String typeCustomer) {
        super(id, name, dateOfBirth, gender, idCard, phoneNumber, email, address);
        this.typeCustomer = typeCustomer;
    }

    public String getTypeCustomer() {
        return typeCustomer;
    }

    public void setTypeCustomer(String typeCustomer) {
        this.typeCustomer = typeCustomer;
    }

    @Override
    public String toString() {
        return "Customer{" +super.toString() +
                "typeCustomer='" + typeCustomer + '\'' +
                '}';
    }
}

File: MODULE_5/ss00_casestudy/not_connect_BE/furama1/angular-furama-app/src/app/model/customer.ts
import {CustomerType} from "src/app/model/customer-type";

export interface Customer {
  id: string,
  name: string,
  gender: string,
  dateOfBirth: string
  idCard: string,
  phone: string,
  email?: string,
  address?: string,
  typeCustomer: string
}
